### ISSUE TITLE
useRouter could use more abstractions

### ISSUE THREAD

**Issue by Seanmclem** [GeekyAnts/unitools#11]
The hand full of abstractions on useRouter https://unitools.geekyants.com/packages/router#props

are kind of bare bones. We have push, replace, back, and navigate?

I'm having to create my own native/web wrappers for things like url params and such. This hook could use more love. It feels forgotten.

**Comment by amrya200t** [comment]
Can you share your solution with us? 😁

**Comment by alex-reusables** [comment]
Yeah I also have this question right now, using the next/router yields a router not mounted error so I'm now somewhat confused about how this all works lol

Edit: ok turns out you get route params as props on the page component kind like:

`export function MyPage({ params }: { params: { id: string } }) {}` so I can access the id in the url if I nest my file like so: 

`/dashboard/location/[id]/page.tsx `
